File: ios/ARWheelApp/Modules/ARLauncherModule.m
// Modules/ARLauncherModule.m
// Objective-C bridge — exposes ARLauncher to React Native JS

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ARLauncher, NSObject)

RCT_EXTERN_METHOD(openARActivity:(NSString *)initialModelPath
                  modelPathsJson:(NSString *)modelPathsJson
                  resolve:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup;

@end

File: ios/ARWheelApp/Modules/ARLauncherModule.swift
// Modules/ARLauncherModule.swift
// React Native native module — presents ARViewController modally.
// Mirrors Android ARLauncherModule.kt exactly (same method signature).

import Foundation
import React
import UIKit

@objc(ARLauncher)
class ARLauncherModule: NSObject {

    @objc static func requiresMainQueueSetup() -> Bool { true }

    @objc func openARActivity(
        _ initialModelPath: String,
        modelPathsJson: String,
        resolve: @escaping RCTPromiseResolveBlock,
        reject: @escaping RCTPromiseRejectBlock
    ) {
        DispatchQueue.main.async {
            guard let presentingVC = RCTPresentedViewController() else {
                reject("NO_CURRENT_ACTIVITY", "No presenting view controller", nil)
                return
            }
            let arVC = ARViewController.create(
                initialModelPath: initialModelPath,
                modelPathsJson: modelPathsJson
            )
            arVC.modalPresentationStyle = .fullScreen
            presentingVC.present(arVC, animated: true) {
                resolve(true)
            }
        }
    }
}
